types/react-redux/index.d.ts

```diff
@@ -1,20 +1,28 @@
 // Type definitions for react-redux 5.0.5
-// Project: https://github.com/reactjs/react-redux
-// Definitions by: Qubo <https://github.com/tkqubo>, Sean Kelley <https://github.com/seansfkelley>, Thomas Hasner <https://github.com/thasner>, Kenzie Togami <https://github.com/kenzierocks>
+// Project: https://github.com/rackt/react-redux
+// Definitions by: Qubo <https://github.com/tkqubo>,
+//                 Sean Kelley <https://github.com/seansfkelley>,
+//                 Thomas Hasner <https://github.com/thasner>,
+//                 Kenzie Togami <https://github.com/kenzierocks>,
+//                 Curits Layne <https://github.com/clayne11>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 2.4
 
 import * as React from 'react';
 import * as Redux from 'redux';
 
 type ComponentClass<P> = React.ComponentClass<P>;
 type StatelessComponent<P> = React.StatelessComponent<P>;
-type Component<P> = ComponentClass<P> | StatelessComponent<P>;
+type Component<P> = React.ComponentType<P>;
 type ReactNode = React.ReactNode;
 type Store<S> = Redux.Store<S>;
 type Dispatch<S> = Redux.Dispatch<S>;
 type ActionCreator<A> = Redux.ActionCreator<A>;
 
+// Diff / Omit taken from https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766
+type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];
```

Comment:
These types seem rather complicated.  Would it make sense to add some sample uses to the test case below?